overlord/snapstate/handlers.go

```diff
@@ -522,6 +527,11 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// Restore configuration for the new revision if available
+	if err = config.RestoreConfigSnapshotMaybe(st, snapsup.Name(), snapsup.Revision()); err != nil {
```

Comment:
> This is actually no-op on new install (we don't have a snapshot of the configuration for the new
> revision, so there is no data to restore, nothing happens), the current configuration remains 
> untouched and is available to the new revision of the snap.

I don't think that's true. Consider this case (and a test for it would be nice):

```
$ ls -l /snap/krita                    
total 0
drwxrwxr-x 8 root root 123 Dez 14 08:55 12
drwxrwxr-x 8 root root 123 Dez 14 18:48 15
drwxrwxr-x 8 root root 123 Set 27  2016 8
lrwxrwxrwx 1 root root   2 Dez 15 15:20 current -> 15

$ sudo snap refresh --revision=12 krita
krita 3.1.0 from 'krita' refreshed

$ ls -l /snap/krita                    
total 0
drwxrwxr-x 8 root root 123 Dez 14 08:55 12
drwxrwxr-x 8 root root 123 Dez 14 18:48 15
drwxrwxr-x 8 root root 123 Set 27  2016 8
lrwxrwxrwx 1 root root   2 Abr  5 00:11 current -> 12
```